### PR TITLE
Fix CI failures by adding apt-get update to Github Actions

### PR DESCRIPTION
**Pull request by embg**
Following advice from [here](https://github.com/actions/virtual-environments/issues/675) and [here](https://serverfault.com/questions/1070642/ubuntu-package-revision-not-found-in-github-actions), add `sudo apt-get update` to hopefully avoid this CI error:
```
E: Failed to fetch http://azure.archive.ubuntu.com/ubuntu/pool/main/g/glibc/libc6-dev-i386_2.31-0ubuntu9.2_amd64.deb  404  Not Found [IP: 52.250.76.244 80]
E: Failed to fetch http://azure.archive.ubuntu.com/ubuntu/pool/main/g/glibc/libc6-x32_2.31-0ubuntu9.2_amd64.deb  404  Not Found [IP: 52.250.76.244 80]
E: Failed to fetch http://azure.archive.ubuntu.com/ubuntu/pool/main/g/glibc/libc6-dev-x32_2.31-0ubuntu9.2_amd64.deb  404  Not Found [IP: 52.250.76.244 80]
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
make[1]: *** [Makefile:337: apt-install] Error 100
make: *** [Makefile:355: libc6install] Error 2
make[1]: Leaving directory '/home/runner/work/zstd/zstd'
Error: Process completed with exit code 2.
```
The first time I ran CI on this commit (on a [PR](https://github.com/embg/zstd/pull/12) in embg/zstd) I got a weird failure on a different job, but the second time everything passed. I propose we land this for now (if it passes CI on facebook/zstd), and if the issue persists there are some other things in the first link above that we can try.

### CODE REVIEW

**Comment by embg** [comment]
Well looks like Travis is failing... will investigate tomorrow.

Edit: [this link](https://travis-ci.community/t/then-sudo-apt-get-update-failed-public-key-is-not-available-no-pubkey-6b05f25d762e3157-in-ubuntu-xenial/1728) should fix the issue, will test tomorrow. Expect some spam from this PR since I can't test this on embg/zstd (Travis CI doesn't give free credits for public repos anymore).

(note to self: Travis debug instructions [here](https://docs.travis-ci.com/user/running-build-in-debug-mode/#legacy-repositories))

Edit: also used [this link](https://techoverflow.net/2018/04/30/how-to-fix-apt-keyexpired-1515625755/) to fix Travis (there were multiple keys that had to be updated).

**Comment by embg** [comment]
Fixed most of the Travis issues, but there is one that apparently [can only be fixed](https://travis-ci.community/t/gpg-errors-with-ubuntugis/8608/2) by moving away from Trusty. So there are two options:
* Move away from Trusty
* Move the "apt-get update" out of the Makefile and have it be specific to Github Actions only (i.e. have Github Actions run "apt-get update" before "make arminstall"). **This is a hack that would allow us to continue using Trusty**, but should only be a temporary solution since we should really always run "apt-get update" before "apt-get install".

Here is the failing Travis job: https://app.travis-ci.com/github/facebook/zstd/jobs/561969471

**Comment by Cyan4973** [comment]
The failing `travis CI` test is a `qemu` `ARM` 32-bit test.
This is the kind of test we should no longer need on Travis, and that we should replace with an equivalent test on Github Action.
The reason it's attached to `Trusty` is likely because we couldn't find a correct combination of flags to correctly compile `zstd` on this platform, resulting in something working fine implicitly on `Trusty` but failing on later platforms due to some changes of default, or something like that.

Bottom line : we would be better off removing that test from `travis CI`, and replace it with an equivalent test on Github Action, that _would not need to be run on `Trusty`_. What matters is `qemu`, and `ARM` 32-bit.

Oh, and btw, it seems we already have such a test : 
https://github.com/facebook/zstd/runs/5410450999?check_suite_focus=true

Just check it's "equivalent enough" to remove the `Trusty` one from `travis CI`.

**Comment by embg** [comment]
> Just check it's "equivalent enough" to remove the `Trusty` one from `travis CI`.

Sounds good, so for the long term solution I'll get rid of the Trusty test as part of https://github.com/facebook/zstd/issues/3011 (sometime this half). Will land the hack fix for now to unbreak CI, and replace with permanent fix once https://github.com/facebook/zstd/issues/3011 is closed.

**Review by felixhandte — APPROVED** [review]
Ship it.
